test/org/sagebionetworks/bridge/services/SchedulePlanServiceTest.java

```diff
@@ -29,40 +33,50 @@
     @Resource
     SchedulePlanServiceImpl schedulePlanService;
     
+    private Study study;
+    
+    @Before
+    public void before() {
+        study = new DynamoStudy();
+        study.setIdentifier("test-study");
+        study.setTaskIdentifiers(Sets.newHashSet("AAA", "BBB", "CCC"));
+    }
+    
     @Test
     public void schedulePlansAreFilteredByAppVersion() {
-        StudyIdentifier studyId = new StudyIdentifierImpl("test-study");
         SchedulePlan savedPlan = null;
         try {
-            SchedulePlan plan = TestUtils.getABTestSchedulePlan(TestConstants.TEST_STUDY);
+            SchedulePlan plan = TestUtils.getABTestSchedulePlan(TEST_STUDY);
             plan.setGuid(null);
             plan.setVersion(null);
             plan.setMinAppVersion(14);
             plan.setMaxAppVersion(15);
-            plan.setStudyKey(studyId.getIdentifier());
-            savedPlan = schedulePlanService.createSchedulePlan(plan);
+            plan.setStudyKey(study.getIdentifier());
+            savedPlan = schedulePlanService.createSchedulePlan(study, plan);
             
-            int baseCount = schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyId).size();
+            int baseCount = schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, study).size();
             
             // This is not high enough for the app version, nothing returned.
             ClientInfo clientInfo = ClientInfo.fromUserAgentCache("TestApp/13 TestSDK/8");
-            List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(clientInfo, studyId);
+            List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(clientInfo, study);
             assertEquals(baseCount-1, plans.size());
             
             // Right in the range, the plan is returned
             clientInfo = ClientInfo.fromUserAgentCache("TestApp/14 TestSDK/8");
-            plans = schedulePlanService.getSchedulePlans(clientInfo, new StudyIdentifierImpl("test-study"));
+            plans = schedulePlanService.getSchedulePlans(clientInfo, study);
             assertEquals(baseCount, plans.size());
             
             clientInfo = ClientInfo.fromUserAgentCache("TestApp/15 TestSDK/8");
-            plans = schedulePlanService.getSchedulePlans(clientInfo, new StudyIdentifierImpl("test-study"));
+            plans = schedulePlanService.getSchedulePlans(clientInfo, study);
             assertEquals(baseCount, plans.size());
             
             // Finally, this is outside the upper-bound, so the plan will not be returned
             clientInfo = ClientInfo.fromUserAgentCache("TestApp/16 TestSDK/40");
-            plans = schedulePlanService.getSchedulePlans(clientInfo, new StudyIdentifierImpl("test-study"));
+            plans = schedulePlanService.getSchedulePlans(clientInfo, study);
             assertEquals(baseCount-1, plans.size());
             
+        } catch(Exception e) {
+            e.printStackTrace();
```

Comment:
This will actually turn test failures into successes. If the problem is the finally block masking test failures, we could move the finally blocks into an @After method, an JUnit *should* show the errors for both.

(canRoundTripSchedulePlan() doesn't fit the pattern, but that test really should be moved to SchedulePlanTest.)